benchmarks/benchmarks/benchmark_dataset.py

```diff
@@ -0,0 +1,164 @@
+# Write the benchmarking functions here.
+# See "Writing benchmarks" in the asv docs for more information.
+import os.path as osp
+import numpy as np
+from tempfile import TemporaryDirectory
+import h5py
+
+class TimeSuite:
+
+    def setup(self):
+        self._td = TemporaryDirectory()
+        path = osp.join(self._td.name, 'test.h5')
+        with h5py.File(path, 'w') as f:
+            f['a'] = np.arange(100000)
+
+        self.f = h5py.File(path, 'r')
+
+    def teardown(self):
+        self.f.close()
+        self._td.cleanup()
+
+    def time_many_small_reads(self):
+        ds = self.f['a']
+        for i in range(10000):
+            arr = ds[i * 10:(i + 1) * 10]
+
+class TimeWritingSuite:
+    """Based on example in GitHub issue 492:
+    https://github.com/h5py/h5py/issues/492
+    """
+    def setup(self):
+        self._td = TemporaryDirectory()
+        path = osp.join(self._td.name, 'test.h5')
+        self.f = h5py.File(path, 'w')
+        self.shape = shape = (128, 1024, 512)
+        self.f.create_dataset(
+            'a', shape=shape, dtype=np.float32, chunks=(1, shape[1], 64)
+        )
+        self.ds = self.f['a']
+        self.data = np.zeros(self.shape[:2])
+        self.datax = np.zeros(self.shape[1:])
+
+    def teardown(self):
+        self.f.close()
+        self._td.cleanup()
+
+    def time_write_index_first_axis(self):
+        for i in range(self.shape[0]):
+            self.ds[i,:,:] = self.datax
+
+    def time_write_slice_first_axis(self):
+        for i in range(self.shape[0]):
+            self.ds[i:i+1,:,:] = self.datax[np.newaxis, ...]
+
+    def time_write_index_last_axis(self):
+        for i in range(self.shape[2]):
+            self.ds[..., i] = self.data
+
+    def time_write_slice_last_axis(self):
+        for i in range(self.shape[2]):
+            self.ds[..., i:i+1] = self.data[..., np.newaxis]
+
+
+class TimeDatasetSuite:
+
+    def setup(self):
+        self._td = TemporaryDirectory()
+        self.path = osp.join(self._td.name, '_dataset.h5')
+        self.f = h5py.File(self.path, 'w')
+
+    def teardown(self):
+        self.f.close()
+        self._td.cleanup()
+
+    def time_create_dataset(self):
+        self.f.create_dataset('dataset', (100,), dtype='f')
+
+    def time_create_empty_dataset(self):
+        self.f.create_dataset('empty', dtyp='f')
```

Comment:
```suggestion
        self.f.create_dataset('empty', dtype='f')
```

Did you try running the benchmarks? I don't think this can have worked.